### ISSUE TITLE
Self compressing neural networks

### ISSUE THREAD

**Issue by msaroufim** [pytorch/ao#658]
Self-Compressing Neural Networks is dynamic quantization-aware training that puts the size of the model in the loss

Paper: https://arxiv.org/pdf/2301.13142
Code: https://github.com/geohot/ai-notebooks/blob/master/mnist_self_compression.ipynb

**Comment by aiepm** [comment]
I'm taking this one.

**Comment by vedant-rupwal** [comment]
Hi everyone! I'm a student in the CodePath AI301 program and I'd love to work on this issue if it is still open and available.

I see there was a mention of it a few months ago, but since it isn't assigned, I'd like to take a crack at reproducing the dynamic QAT behavior described in the paper locally this week. Any pointers on where to start integrating this within the module: qat codebase would be greatly appreciated!
